app/views/course/assessment/answer/programming/_annotations.json.jbuilder

```diff
@@ -1,13 +1,16 @@
 # frozen_string_literal: true
-
 json.annotations programming_files do |file|
   json.fileId file.id
   json.topics(file.annotations.reject { |a| a.discussion_topic.post_ids.empty? }) do |annotation|
     topic = annotation.discussion_topic
     next unless can_grade || !topic.posts.only_published_posts.empty?
 
     json.id topic.id
-    json.postIds topic.post_ids
+    if can_grade
+      json.postIds topic.post_ids
+    else
+      json.postIds topic.posts.only_published_posts.ids
```

Comment:
Does this trigger N+1 queries for each annotation, to get `annotation.discussion_topic.posts.only_published_posts.ids` or does the data already contain the workflowstate (published) property?

Reply:
The latter. `topic` already has all the information, including the `published` property